data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/admin/AddPeerTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Consensys Software Inc., 2025
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beaconrestapi.handlers.tekuv1.admin;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_INTERNAL_SERVER_ERROR;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
+
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.beaconrestapi.AbstractMigratedBeaconHandlerTest;
+import tech.pegasys.teku.networking.p2p.discovery.DiscoveryNetwork;
+
+class AddPeerTest extends AbstractMigratedBeaconHandlerTest {
+
+  private final DiscoveryNetwork<?> discoveryNetwork = mock(DiscoveryNetwork.class);
+
+  @BeforeEach
+  public void setup() {
+    setHandler(new AddPeer(network));
+  }
+
+  @Test
+  public void shouldReturnOkWhenAValidListIsProvided() throws Exception {
+    final List<String> peers =
+        List.of(
+            "/ip4/178.128.136.233/udp/9001/p2p/16Uiu2HAmNSjEXNqaFjfLePKk87WZ7QwuTPd1HPEfJEeYnaC3bGZ1");
+    request.setRequestBody(peers);
+    when(network.getDiscoveryNetwork()).thenReturn(Optional.of(discoveryNetwork));
+    handler.handleRequest(request);
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+  }
+
+  @Test
+  public void shouldReturnBadRequestIfInvalidPeerAddress() throws Exception {
```

Comment:
should add the standard tests on response codes eg. `metadata_shouldHandle400` etc for all codes. these will exercise the metadata without the handler.